### DISCUSSION TITLE
Are pingInterval and pingTimeout are equivalents of Keep-Alive?

### DISCUSSION THREAD

**Discussion by awesomelike** [socketio/socket.io#4098] [Q&A] [answered]
I am wondering if the options pingInterval and pingTimeout are for Keep-Alive functionality? I mean, are they for keeping the idle connection unclosed (and not getting 504 Gateway Timeout errors)?

**Comment by darrachequesne** [comment, marked as answer]
No, they are meant to make sure the connection between the client and the server is still up and running.

As a side-effect, it indeed prevents 504 Gateway timeout errors too.

**Reply by awesomelike** [reply]
Thank you very much! Now with this, I know how to prevent 504 Gateway timeout errors :)
